scripts/execute_cypress.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+# Exit immediately if a command exits with a non-zero status,
+# Treat unset variables as an error, and prevent errors in a pipeline from being masked
+set -euo pipefail
+
+# Initialize tmp_file globally
+tmp_file=""
+
+# Define arrays for services, etc.
+# Read service arrays from environment variables
+read -r -a payments <<< "${PAYMENTS[@]:-}"
+read -r -a payouts <<< "${PAYOUTS[@]:-}"
+read -r -a payment_method_list <<< "${PAYMENT_METHOD_LIST[@]:-}"
+read -r -a routing <<< "${ROUTING[@]:-}"
+
+# Define arrays
+connector_map=()
+failed_connectors=()
+
+# Define an associative array to map environment variables to service names
+declare -A services=(
+  ["PAYMENTS"]="payments"
+  ["PAYOUTS"]="payouts"
+  ["PAYMENT_METHOD_LIST"]="payment_method_list"
+  ["ROUTING"]="routing"
+)
+
+# Function to print messages in color
+function print_color() {
+  # Input params
+  local color="$1"
+  local message="$2"
+
+  # Define colors
+  local reset='\033[0m'
+  local red='\033[0;31m'
+  local green='\033[0;32m'
+  local yellow='\033[0;33m'
+
+  # Use indirect reference to get the color value
+  echo -e "${!color}${message}${reset}"
+}
+export -f print_color
```

Comment:
`parallel` runs commands in a `subshell` and it does not have to access to the function that reside outside of the script. 
so we need to export it such that it is accessible by the function used by the subshell. in this case, `execute_test` makes use of `print_color` within the subshell and hence we need it to be exported.

Reply:
Do you need all the functions to be exported though?

Reply:
yes, we need all the functions that are used by the subshell to be exported: 

<img width="375" alt="image" src="https://github.com/user-attachments/assets/1a561e8a-c053-469b-bc9d-73e7237073c4">

we're only exporting 2 functions, namely:

- `print_color`
- `execute_test`

`execute_test` is called by `parallel`. since the command is run in subshell, we need to `export` it as a function and this `execute_test` requires `print_color` and that also needs to be exported

Reply:
`read_service_arrays` seems to have been exported as well?

Reply:
yes, at `L60`.